src/suggestEnvironmentVars.ts

```diff
@@ -48,43 +52,69 @@ function createSuggestion(
+  const openingBraceIndex = currentLine.lastIndexOf("${", cursorIndex);
+  const closingBraceIndex = currentLine.indexOf("}", cursorIndex);
+  return [openingBraceIndex, closingBraceIndex];
+}
+
+
+function createSuggestionsArray(
+  envVarsObj: Record<string, { src: string; val: string }>,
+  position: vscode.Position
+): vscode.CompletionItem[] {
+
+  // Convert the object into a Map for easier iteration
+  const envVars = new Map(Object.entries(envVarsObj));
+
+  // Map each entry in the Map to a completion suggestion
+  const suggestionsArray = Array.from(envVars.entries()).map(([env, meta]) => {
+    const suggestion = createSuggestion(env, meta.val, meta.src, position);
+    return suggestion;
+  });
+
+  return suggestionsArray;
+}
+
+
+function getTriggerPrefixAndPostfix(currentLine: string, cursorPosition: number): [boolean, boolean] {
+  const triggerPrefix = currentLine.substring(0, cursorPosition).includes("${");
+  const triggerPostfix = currentLine.substring(cursorPosition).includes("}");
+  return [triggerPrefix, triggerPostfix];
+}
+
 export function suggestENVs() {
   return vscode.languages.registerCompletionItemProvider(
     { language: "lama2", scheme: "file" },
     {
-      // eslint-disable-next-line no-unused-vars
-      provideCompletionItems(document, position, token, context) {
+      provideCompletionItems(document, position) {
         const currentLine = document.lineAt(position.line).text;
-        const triggerPrefix = currentLine
-          .substring(0, position.character)
-          .includes("${");
-        const triggerPostfix = currentLine
-          .substring(position.character)
-          .includes("}");
+        const cursorIndex = position.character;
 
-        const envVarsObj = getEnvsFromEnvCommand() as Record<
-          string,
-          { src: string; val: string }
-        >; // Explicitly cast to the correct type
-        const envVars = new Map(Object.entries(envVarsObj));
+        const [openingBraceIndex, closingBraceIndex] = getBraceIndicesOfCurLine(currentLine, cursorIndex);
+        const isInsidePlaceholder = isCursorInsidePlaceholder(openingBraceIndex, closingBraceIndex)
 
-        const suggestionsArray = Array.from(envVars.entries()).map(
-          ([env, meta]) => createSuggestion(env, meta.val, meta.src, position)
-        );
+        if (isInsidePlaceholder) {
+          const typedEnvArg = currentLine.substring(openingBraceIndex + 2, cursorIndex);
+          const envVarsObj = getEnvsFromEnvCommand(typedEnvArg);
+          const suggestionsArray = createSuggestionsArray(envVarsObj, position);
+          const [triggerPrefix, triggerPostfix] = getTriggerPrefixAndPostfix(currentLine, cursorIndex);
 
-        if (triggerPrefix && !triggerPostfix) {
-          return suggestionsArray;
-        } else if (triggerPrefix) {
-          return suggestionsArray;
-        } else {
-          return [];
+          if (triggerPrefix && !triggerPostfix) {
```

Comment:
The second part doesn't matter here. If `triggerPostfix` is true, we return `suggestionsArray`. If it is `false`, still we do the same.

Reply:
Right